tfx/examples/cifar10/cifar10_utils_native_keras.py

```diff
@@ -0,0 +1,338 @@
+# Lint as: python2, python3
+# Copyright 2019 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Python source file includes CIFAR10 utils for Keras model.
+
+The utilities in this file are used to build a model with native Keras.
+This module file will be used in Transform and generic Trainer.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from typing import List, Text
+
+import os
+import absl
+import tensorflow as tf
+import tensorflow_transform as tft
+
+import flatbuffers
+# pylint: disable=g-direct-tensorflow-import
+from tflite_support import metadata as _metadata
+from tflite_support import metadata_schema_py_generated as _metadata_fb
+# pylint: enable=g-direct-tensorflow-import
+
+from tfx.components.trainer.rewriting import converters
+from tfx.components.trainer.rewriting import rewriter
+from tfx.components.trainer.rewriting import rewriter_factory
+
+from tfx.components.trainer.executor import TrainerFnArgs
+
+# When training on the whole dataset use following numbers instead
+# _TRAIN_DATA_SIZE = 50000
+# _EVAL_DATA_SIZE = 10000
+# _TRAIN_BATCH_SIZE = 64
+# _EVAL_BATCH_SIZE = 64
+
+_TRAIN_DATA_SIZE = 100
+_EVAL_DATA_SIZE = 100
+_TRAIN_BATCH_SIZE = 32
+_EVAL_BATCH_SIZE = 32
+
+IMAGE_KEY = 'image'
+LABEL_KEY = 'label'
+
+def transformed_name(key):
+  return key + '_xf'
+
+def _gzip_reader_fn(filenames):
+  """Small utility returning a record reader that can read gzip'ed files."""
+  return tf.data.TFRecordDataset(filenames, compression_type='GZIP')
+
+def _get_serve_image_fn(model):
+  """Returns a function that feeds the input tensor into the model."""
```

Comment:
Lets make this a bit more clear. It takes an input tensor representing the image. It expects that there are 3 channels...